core/consensus/authority/impl/authority_manager_impl.cpp

```diff
@@ -509,21 +509,55 @@ namespace kagome::authority {
     };
 
     KAGOME_PROFILE_START(is_ancestor_node_finalized)
-
-    /// TODO(Harrm): Should account for finality when fetching an authority set
-    /// for some purposes, but not when scheduling further changes
-    IsBlockFinalized is_ancestor_node_finalized = true;
+    IsBlockFinalized is_ancestor_node_finalized =
+        ancestor_node->current_block == block_tree_->getLastFinalized();
+    or directChainExists(ancestor_node->current_block,
+                         block_tree_->getLastFinalized(););
     KAGOME_PROFILE_END(is_ancestor_node_finalized)
 
+    std::optional<std::shared_ptr<const primitives::AuthoritySet>> maybe_set =
```

Comment:
Please comment about maybe_set, this logic is unclear.

Reply:
Added